src/behavior-considered-undefined.md

```diff
@@ -34,6 +34,7 @@ code.
 * Invoking undefined behavior via compiler intrinsics.
 * Executing code compiled with platform features that the current platform
   does not support (see [`target_feature`]), *except* if the platform explicitly documents this to be safe.
+* Performing non-nop coercion on a dangling or unaligned raw pointer.
```

Comment:
I'm not exactly clear why this is being added in this PR, since I can't find any reference to it related to RFC 3324. Can you say more about why this is being added? Is there some context here? 

cc @RalfJung since you often know about UB stuff.

Reply:
I believe it was decided in a T-lang FCP in https://github.com/rust-lang/rust/issues/101336, the stabilization of trait_upcasting just becomes the first time that this behavior can be exercised, as far as I recall.

Reply:
My expectation was that it is completely okay for the pointer itself to be dangling or unaligned; only the vtable must be valid. And we already require the vtable to be valid; we say it is UB when one creates the following:
```
Invalid metadata in a wide reference, `Box<T>`, or raw pointer
```

So I don't think any change is needed to the UB list. At least I highly doubt we want to require the data part of the wide raw pointer to be valid in any way for such a coercion.

Cc @rust-lang/opsem